sources/react-app/src/pages/Post.js

```diff
@@ -14,22 +14,160 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import BaseLayout from '../shared/BaseLayout';
 import ContentType from '../shared/ContentType';
-import PopularPostsAside from '../shared/PopularPostsAside';
+import RecentPostsAside from '../shared/RecentPostsAside';
 import PostCard, { IMAGE_BACKGROUND } from '../shared/PostCard';
 import SidebarCategories from '../shared/SidebarCategories';
 import SidebarTags from '../shared/SidebarTags';
 import SidebarSearch from '../shared/SidebarSearch';
 import SidebarBios from '../shared/SidebarBios';
 import DropZone from '../shared/DropZone';
+import { fetchQuery } from '../relayEnvironment';
+import { parseDescriptor } from '@craftercms/content';
 
 export default function (props) {
-  const { model, posts } = props;
+  const {
+    model,
+    posts,
+    meta: {
+      siteTitle
+    }
+  } = props;
   const modelPath = model.craftercms.path;
+  const [related, setRelated] = useState();
+  const categoriesKeys = model.categories_o?.map(category => `{matches: "${category.key}"}`);
+  const tagsKeys = model.tags_o?.map(tag => `{matches: "${tag.key}"}`);
+
+  useEffect(() => {
```

Comment:
pagination and limit.
refactor this as hook so you can use same query for all listings. possibly even same hook as recent?